src/Compilers/Core/Portable/Emit/ErrorType.cs

```diff
@@ -174,53 +174,23 @@ string Cci.INamedEntity.Name
         /// <summary>
         /// A fake containing assembly for an ErrorType object.
         /// </summary>
-        private class ErrorAssembly : Cci.IAssemblyReference
+        private sealed class ErrorAssembly : Cci.IAssemblyReference
         {
             public static readonly ErrorAssembly Singleton = new ErrorAssembly();
+            
             /// <summary>
             /// For the name we will use a word "Error" followed by a guid, generated on the spot.
             /// </summary>
-            private static readonly string s_name = "Error" + Guid.NewGuid().ToString("B");
+            private static readonly AssemblyIdentity s_identity = new AssemblyIdentity(
+                name: "Error" + Guid.NewGuid().ToString("B"),
```

Comment:
Not sure. The comment at the use site of this singleton says

```
// Substitute error types with a special singleton object.
// Unreported bad types can come through NoPia embedding, for example.
```

@AlekseyTs ?